### PR TITLE
Feature/buttons bar

### PR DESCRIPTION
**Pull request by DanBotero**
bar-buttons in the same place and working.

### CODE REVIEW

**Comment by kirjs** [comment]
Please push the latest!

**Comment by DanBotero** [comment]
Buttons bar 
screen shot 2017-04-30 at 1 23 06 pm

File: src/app/feedback/feedback.module.ts [hunk 1]
```diff
@@ -22,14 +17,16 @@ export const af = AngularFireModule.initializeApp(firebaseConfig);
   imports: [
     CommonModule,
     ReactiveFormsModule,
-    af
+    af,
+    AngularFireModule.initializeApp(firebaseConfig),
```
Comment (prestonvanloon) [inline src/app/feedback/feedback.module.ts]: This is initializing the app twice. Please remove.

File: src/app/demo/bootstrap/bootstrap.module.ts [hunk 2]
```diff
@@ -5,9 +5,10 @@ import {SlidesRoutes} from '../../presentation/slide-routes';
 
 import {ExerciseModule} from '../../exercise/exersice.module';
 import {PresentationModule} from '../../presentation/presentation.module';
-import {FeedbackModule} from "../../feedback/feedback.module";
 import {BrowserWindowModule} from '../../browser-window/browser-window.module';
 
+import {FeedbackModule} from "../../feedback/feedback.module";
```
Comment (prestonvanloon) [inline src/app/demo/bootstrap/bootstrap.module.ts]: Why move this import?

File: src/app/demo/component-tree/component-tree.component.html [hunk 3]
```diff
@@ -21,40 +26,46 @@
...
-          <h3 style="text-align: justify;">Parent Component</h3>
-          <app-code-editor [fontSize]="15" [code]="code.parentComponentSkeleton.code"></app-code-editor>
-        </div>
-        <div class="col-6" style="margin-left: 10px;">
-          <h3>Child Component</h3>
-          <app-code-editor [fontSize]="15" [code]="code.childComponentSkeleton.code"></app-code-editor>
-        </div>
+    <h2>Generate your components</h2>
+    <div style="display: flex; flex-direction: row;">
+      <div style="width: 50%">
```
Comment (prestonvanloon) [inline src/app/demo/component-tree/component-tree.component.html]: Please, no inline styles.

File: src/app/demo/component-tree/component-tree.module.ts [hunk 4]
```diff
@@ -7,7 +7,7 @@ import {ExerciseModule} from '../../exercise/exersice.module';
 import {PresentationModule} from '../../presentation/presentation.module';
 import {FeedbackModule} from '../../feedback/feedback.module';
 import {BrowserWindowModule} from '../../browser-window/browser-window.module';
-import {Ng2TsExercises} from '../../../../ng2ts/ng2ts';
+import {Ng2TsExercises} from "../../../../ng2ts/ng2ts";
```
Comment (prestonvanloon) [inline src/app/demo/component-tree/component-tree.module.ts]: Revert to single quotes

File: src/app/demo/dependency-injection/dependency-injection.module.ts [hunk 5]
```diff
@@ -1,19 +1,21 @@
...
 import {DependencyInjectionComponent} from './dependency-injection.component';
 import {RouterModule} from '@angular/router';
 import {SlidesRoutes} from '../../presentation/slide-routes';
-import {FeedbackModule} from "../../feedback/feedback.module";
+
 import {ExerciseModule} from '../../exercise/exersice.module';
 import {PresentationModule} from '../../presentation/presentation.module';
-import {Ng2TsExercises} from '../../../../ng2ts/ng2ts';
+import {Ng2TsExercises} from "../../../../ng2ts/ng2ts";
+import {FeedbackModule} from "../../feedback/feedback.module";
```
Comment (prestonvanloon) [inline src/app/demo/dependency-injection/dependency-injection.module.ts]: Why move this module?

Comment (prestonvanloon) [inline src/app/demo/dependency-injection/dependency-injection.module.ts]: Single quotes please

File: src/app/demo/typescript/typescript.module.ts [hunk 6]
```diff
@@ -4,8 +4,8 @@ import {RouterModule} from '@angular/router';
 import {SlidesRoutes} from '../../presentation/slide-routes';
 import {ExerciseModule} from '../../exercise/exersice.module';
 import {PresentationModule} from 'app/presentation/presentation.module';
-import {FeedbackModule} from '../../feedback/feedback.module';
-import {CommonModule} from '@angular/common';
+import {FeedbackModule} from "../../feedback/feedback.module";
+import {CommonModule} from "@angular/common";
```
Comment (prestonvanloon) [inline src/app/demo/typescript/typescript.module.ts]: Single quotes.

File: src/app/feedback/feedback-widget/feedback-widget.component.ts [hunk 7]
```diff
@@ -1,23 +1,11 @@
...
+import {Component, OnInit, ElementRef, OnDestroy} from '@angular/core';
+import {AngularFire, FirebaseListObservable} from 'angularfire2';
+
+import {Subscription} from "rxjs/Subscription";
 import { Message } from './../message';
-import { Observable } from 'rxjs/Observable';
-import { PresentationComponent } from '../../presentation/presentation/presentation.component';
-import { Subscription } from 'rxjs/Subscription';
+import {FormBuilder, FormGroup, Validators} from "@angular/forms";
+import {ActivatedRoute, Router} from "@angular/router";
```
Comment (prestonvanloon) [inline src/app/feedback/feedback-widget/feedback-widget.component.ts]: Single quotes

File: src/app/feedback/feedback-widget/feedback-widget.component.ts [hunk 8]
```diff
@@ -88,6 +76,18 @@ export class FeedbackWidgetComponent implements OnInit, OnDestroy {
     this.initialized = true;
   }
 
+  buttonClicked() {
+    this.open = !this.open;
+    document.addEventListener('click', () => {
```
Comment (prestonvanloon) [inline src/app/feedback/feedback-widget/feedback-widget.component.ts]: Is this going to add an event listener every time the button is clicked? Why?

**Comment by kirjs** [comment]
@prestonvanloon PTAL

File: src/app/presentation/menu-shortcut/menu-shortcut.component.ts [hunk 9]
```diff
@@ -15,8 +15,14 @@ export class MenuShortcutComponent implements OnInit {
 
   ngOnInit() {
   }
+
   onDisplay() {
-    this.onOpenMenu = (this.onOpenMenu === false) ? true : false;
+    this.onOpenMenu = !this.onOpenMenu;
+    document.addEventListener('click', (event) => {
```
Comment (prestonvanloon) [inline src/app/presentation/menu-shortcut/menu-shortcut.component.ts]: This adds a global event listener everytime onDisplay is called, right? I don't think that is desired. 

Can you use angular style `(click)="handleMenuButtonClick()"`? If not, please leave a detailed comment here and in the other eventListener you are creating to explain why.

File: src/app/feedback/feedback.module.ts (hunk 1 above)
Comment (kirjs) [inline src/app/feedback/feedback.module.ts]: Done

File: src/app/feedback/feedback-widget/feedback-widget.component.ts (hunk 8 above)
Comment (kirjs) [inline src/app/feedback/feedback-widget/feedback-widget.component.ts]: moved to host listener

File: src/app/presentation/menu-shortcut/menu-shortcut.component.ts (hunk 9 above)
Comment (kirjs) [inline src/app/presentation/menu-shortcut/menu-shortcut.component.ts]: done

**Comment by kirjs** [comment]
@prestonvanloon I fixed most of it, lmk if looks good and I'll merge

File: src/app/demo/dependency-injection/dependency-injection.module.ts [hunk 10]
```diff
@@ -1,19 +1,20 @@
...
+import {SlidesRoutes} from '../../presentation/slide-routes';
 
 
 const routes = RouterModule.forChild(
   SlidesRoutes.get(DependencyInjectionComponent)
 );
 
 @NgModule({
-  imports: [routes, PresentationModule, ExerciseModule, FeedbackModule],
+  imports: [routes, PresentationModule, ExerciseModule,FeedbackModule],
```
Comment (prestonvanloon) [inline src/app/demo/dependency-injection/dependency-injection.module.ts]: Revert deleted space

**Pull request merged**